File: 30.maximum-gap.cpp
class Solution {
public:
    int maximumGap(vector<int>& nums) {
        if(nums.size()<2)return 0;
        sort(nums.begin(),nums.end());
        int len_num = nums.size();
         int max_sum=0;
            for(int i=0,j=i+1;i<len_num-1,j<len_num;i++,j++){
                max_sum = max(max_sum,abs(nums[i]-nums[j]));
        }
        return max_sum;
    }
};

File: 30.maximum-gap.py
class Solution:
    def maximumGap(self, nums: List[int]) -> int:
        
        len_nums=len(nums)
        if len_nums<2: 
            return 0
        nums.sort()
        max_=0
        i=0
        j=i+1
        while i<len_nums-1:
            max_ = max(max_,abs(nums[i]-nums[j]))
            i+=1
            j+=1
        return max_
    
